pallets/api/src/extension.rs

```diff
@@ -0,0 +1,131 @@
+use core::{fmt::Debug, marker::PhantomData};
+use frame_support::traits::Get;
+pub use pop_chain_extension::{
+	Config, DecodingFailed, DispatchCall, ReadState, Readable, DECODING_FAILED_ERROR,
+};
+use pop_chain_extension::{
+	Converter, Decodes, Environment, LogTarget, Matches, Processor, Result, RetVal,
+};
+use sp_runtime::DispatchError;
+use sp_std::vec::Vec;
+
+/// The logging target for the chain extension.
+pub const LOG_TARGET: &str = "pop-api::extension";
+
+/// The chain extension used by the API.
+pub type Extension<Functions> = pop_chain_extension::Extension<Functions>;
+/// Decodes output by prepending bytes from ext_id() + func_id()
+pub type DecodesAs<Output, Error, Logger = ()> = Decodes<Output, Error, Prepender, Logger>;
+
+/// Prepends bytes from ext_id() + func_id() to prefix the encoded input bytes to determine the versioned output
+pub struct Prepender;
+impl Processor for Prepender {
+	/// The type of value to be processed.
+	type Value = Vec<u8>;
+
+	/// The log target.
+	const LOG_TARGET: &'static str = "pop-api::extension::processor";
+
+	/// Processes the provided value.
+	///
+	/// # Parameters
+	/// - `value` - The value to be processed.
+	/// - `env` - The current execution environment.
+	fn process(mut value: Self::Value, env: &impl Environment) -> Self::Value {
+		// Resolve version, pallet and call index from environment
+		let version = version(env);
+		let (module, index) = module_and_index(env);
+		// Prepend bytes
+		value.insert(0, version);
+		value.insert(1, module);
+		value.insert(2, index);
```

Comment:
These insertions would be more efficient with using splice():
```rust
    value.splice(0..0, [index, module, version]);
```

Reply:
See https://github.com/r0gue-io/pop-node/pull/265

Reply:
awesome! Those lines were bothering for a couple PRs 😅